File: Classes/EVEOnlineAPI/Character/EVEResearch.m
//
//  EVEResearch.m
//  EVEOnlineAPI
//
//  Created by Artem Shimanski on 6/4/10.
//  Copyright 2010 __MyCompanyName__. All rights reserved.
//

#import "EVEResearch.h"


@implementation EVEResearchItem

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"agentID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"skillTypeID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"researchStartDate":@{@"type":@(EVEXMLSchemePropertyTypeDate)},
				   @"pointsPerDay":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"remainderPoints":@{@"type":@(EVEXMLSchemePropertyTypeScalar)}};
	
	return scheme;
}

@end


@implementation EVEResearch

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"research":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEResearchItem class]}};
	
	return scheme;
}

@end

File: EVEAPI/EVEAPI/EVEResearch.swift
//
//  EVEResearch.swift
//  EVEAPI
//
//  Created by Artem Shimanski on 29.11.16.
//  Copyright © 2016 Artem Shimanski. All rights reserved.
//

import UIKit

public class EVEResearchItem: EVEObject {
	public var agentID: Int64 = 0
	public var skillTypeID: Int = 0
	public var researchStartDate: Date = Date.distantPast
	public var pointsPerDay: Double = 0
	public var remainderPoints: Double = 0
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
		"agentID":EVESchemeElementType.Int64(elementName:nil, transformer:nil),
		"skillTypeID":EVESchemeElementType.Int(elementName:nil, transformer:nil),
		"researchStartDate":EVESchemeElementType.Date(elementName:nil, transformer:nil),
		"pointsPerDay":EVESchemeElementType.Double(elementName:nil, transformer:nil),
		"remainderPoints":EVESchemeElementType.Double(elementName:nil, transformer:nil),
		]
	}
}

public class EVEResearch: EVEResult {
	public var research: [EVEResearchItem] = []
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"research":EVESchemeElementType.Rowset(elementName: nil, type: EVEResearchItem.self, transformer: nil),
		]
	}
}
